src/newgrf_airport.h

```diff
@@ -130,8 +130,8 @@ struct AirportSpec {
 	/** Get the index of this spec. */
 	uint8_t GetIndex() const
 	{
-		assert(this >= specs && this < endof(specs));
-		return (uint8_t)(this - specs);
+		assert(this >= std::begin(specs) && this < std::end(specs));
+		return static_cast<uint8_t>(std::distance(this, std::cbegin(specs)));
```

Comment:
It's fine either way, but it feels a little odd to have `std::distance(end, start)`

```suggestion
		return static_cast<uint8_t>(std::distance(std::cbegin(specs), this));
```